src/main/java/ecommerce/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,67 @@
+package ecommerce.filters;
+
+import ecommerce.lib.Injector;
+import ecommerce.model.Role;
+import ecommerce.model.User;
+import ecommerce.service.interfaces.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorizationFilter implements Filter {
+    private static final Injector injector = Injector.getInstance("ecommerce");
+    private static final String USER_ID = "user_id";
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+    private final Map<String, Set<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/user/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/order/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/add", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/complete", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/product/buy", Set.of(Role.RoleName.USER));
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String requestedUrl = req.getServletPath();
+        if (protectedUrls.get(requestedUrl) == null) {
```

Comment:
let's put both conditions in one `if`
also, we can use `containsKey()` here

Reply:
UPD: to make it work with one `if` you will have to remove `user` local variable, so this is optional

Reply:
so I can leave as it is?

Reply:
refactored a bit, using containsKey()